test/handlers/connect.js

```diff
@@ -13,12 +13,16 @@ test('reject clients with no clientId running on MQTT 3.1', function (t) {
   }
 
   client.broker = broker
+  client.conn = {
+    destroy: function () {}
```

Comment:
Why this change?

Reply:
`handleConnect` is a sub handler in `handle`, and there's part of `connect` login in `handle` function. It is best to emulate much more real world case.

Reply:
If this is called, would you mind placing an assertion in there?